build.gradle

```diff
@@ -235,76 +236,99 @@ java {
     withSourcesJar()
 }
 
-minecraft.accessTransformers.files(
-        file('src/main/resources/META-INF/accesstransformer.cfg'),
-        file('src/additions/resources/META-INF/accesstransformer.cfg'),
-        //Dev time only ATs so the file name doesn't have to match accesstransformer.cfg
-        file('src/datagen/main/resources/META-INF/datagen_ats.cfg'),
-        file('src/gameTest/resources/META-INF/gametest_ats.cfg')
-)
-
-runs {
-    configureEach {
-        if (hasProperty('forge_force_ansi')) {
-            //Force ansi if declared as a gradle variable, as the auto detection doesn't detect IntelliJ properly
-            // or eclipse's plugin that adds support for ansi escape in console
-            systemProperties.put('terminal.ansi', (String) property('forge_force_ansi'))
-        }
-
-        modSources.add((SourceSet[]) [sourceSets.main, sourceSets.api])
-
-        for (String name : secondaryModules) {
-            modSources.add((SourceSet) sourceSets.named(name).get())
-        }
+neoForge {
+    // This enables modding tasks and configurations
+    enable {
+        version = forge_version
+        enabledSourceSets = sourceSets // All source sets use Minecraft code
+    }
 
-        //if the selected toolchain is a JBR, enable DCEVM
-        if (javaToolchains.launcherFor(java.toolchain).map { it.metadata.vendor }.getOrElse('').contains('JetBrains')) {
-            jvmArguments.add('-XX:+AllowEnhancedClassRedefinition')
+    mods {
+        mekanism {
+            modSourceSets = [sourceSets.api, sourceSets.main, sourceSets.datagenMain]
```

Comment:
If I understand this properly, this should not include datagenMain and instead should be as a separate mod similar to what you did for projecte. Given we don't want the datagen code applying for normal runs in case it causes things to break, or if we want to run the client before we finished porting the datagen stuff to the new MC version.